lib/utils/jsontools.go

```diff
@@ -114,8 +114,26 @@ func FastMarshalIndent(v interface{}, prefix, indent string) ([]byte, error) {
 	return data, nil
 }
 
-// WriteJSON marshals multiple documents as a JSON list with indentation.
-func WriteJSON(w io.Writer, values interface{}) error {
+// WriteJSONArray marshals values as a JSON array.
+func WriteJSONArray[T any](w io.Writer, values []T) error {
+	if len(values) == 0 {
+		_, err := w.Write([]byte("[]"))
```

Comment:
Rather that writing a string/bytes directly, perhaps just let `json.Marshal` do it. `null` is emitted for nil slices, and `[]` for empty non-nil slices (see https://go.dev/play/p/MUAis-9xFXn). So perhaps just convert it to a non-nil slice if nil?

```
if values == nil {
    values = []T{}
}
return writeJSON(w, values)
```